### PR TITLE
Add Unified-Plan API.

### PR DESCRIPTION
**Pull request by cloudwebrtc**
Add files:
- rtc_rtcp_parameters.dart
- rtc_rtp_parameters.dart
- rtc_rtp_receiver.dart
- rtc_rtp_sender.dart
- rtc_rtp_transceiver.dart
- rtc_track_event.dart

new features:
- Simulcast/SVC
- replaceTrack

### CODE REVIEW

File: lib/src/rtc_peerconnection.dart [hunk 1]
```diff
@@ -361,6 +383,116 @@ class RTCPeerConnection {
...
+    } on PlatformException catch (e) {
+      throw 'Unable to RTCPeerConnection::createSender: ${e.message}';
+    }
+  }
+
+  Future<RTCRtpSender> addTrack(MediaStreamTrack track,
+      [List<String> streamIds]) async {
+    try {
+      final response =
+          await _channel.invokeMethod('addTrack', <String, dynamic>{
```
Comment (tomsun) [inline lib/src/rtc_peerconnection.dart:413]: addTrack missing on iOS currently?

Comment (cloudwebrtc) [reply lib/src/rtc_peerconnection.dart:413]: We are currently writing the Android native adaptation code, and we will write iOS code later, and we will eventually merge them all together.

**Comment by arttobe** [comment]
Is there a date when it will be merged

**Comment by cloudwebrtc** [comment]
@arttobe About three weeks.

File: lib/src/web/rtc_peerconnection_impl.dart [hunk 2]
```diff
@@ -83,6 +83,12 @@ class RTCPeerConnectionWeb extends RTCPeerConnection {
       onSignalingState?.call(_signalingState);
     });
 
+    js.JsObject.fromBrowserObject(_jsPc)['connectionstatechange'] =
+        js.JsFunction.withThis((_, state) {
+      _connectionState = peerConnectionStateForString(state);
+      onConnectionState.call(_connectionState);
```
Comment (wer-mathurin) [inline lib/src/web/rtc_peerconnection_impl.dart:89]: Missing null aware operator => onConnectionState?

File: lib/src/web/rtc_peerconnection_impl.dart [hunk 3]
```diff
@@ -265,6 +275,9 @@ class RTCPeerConnectionWeb extends RTCPeerConnection {
   @override
   Future<RTCRtpSender> addTrack(MediaStreamTrack track,
       [List<MediaStream> streams]) {
+    var _track = track as MediaStreamTrackWeb;
+    var _stream = streams[0] as MediaStreamWeb;
+    _jsPc.addTrack(_track.jsTrack, _stream.jsStream);
     // TODO: implement addTrack
     throw UnimplementedError();
```
Comment (wer-mathurin) [inline lib/src/web/rtc_peerconnection_impl.dart:282]: This method is half implemented.
And probably, we need to add the track to all the streams,  Maybe a for a for loop ? 
@rostopira can you comment on this?

File: lib/src/web/rtc_peerconnection_impl.dart (hunk 2 above)
Comment (cloudwebrtc) [reply lib/src/web/rtc_peerconnection_impl.dart:89]: Yeah, I missed `?`.

File: lib/src/web/rtc_peerconnection_impl.dart (hunk 3 above)
Comment (cloudwebrtc) [reply lib/src/web/rtc_peerconnection_impl.dart:282]: Regarding the js part, I wonder if there is a better way to implement it.
like https://pub.dev/packages/js.

Comment (cloudwebrtc) [reply lib/src/web/rtc_peerconnection_impl.dart:282]: Another idea is to use js/js.dart to package directly from the browser environment, but I am not sure that it is effective.

like this: https://pub.dev/packages/import_js_library.

Comment (cloudwebrtc) [reply lib/src/web/rtc_peerconnection_impl.dart:282]: Yeah, Perhaps it is the best choice at this stage.

**Comment by wer-mathurin** [comment]
LGTM

**Review by wer-mathurin — APPROVED** [review]

**Pull request merged**